File: examples/src/Shapes/ConnectorLineAngle.cs
using System.IO
import aspose.slides as slides
using System

namespace Aspose.slides.Examples.CSharp.shapes
{
    public class ConnectorLineAngle
    {
        #ExStart:ConnectorLineAngle
        public static void Run()
        {
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Shapes()

            Presentation pres = new Presentation(dataDir + "ConnectorLineAngle.pptx")
            Slide slide = (Slide)pres.slides[0]
            Shape shape
            for (i = 0 i < slide.shapes.Count i++)
            {
                double dir = 0.0
                shape = (Shape)slide.shapes[i]
                if (shape is AutoShape)
                {
                    AutoShape ashp = (AutoShape)shape
                    if (ashp.ShapeType == slides.ShapeType.LINE)
                    {
                        dir = getDirection(ashp.width, ashp.height, Convert.ToBoolean(ashp.frame.flip_h), Convert.ToBoolean(ashp.frame.flip_v))
                    }
                }
                else if (shape is Connector)
                {
                    Connector ashp = (Connector)shape
                    dir = getDirection(ashp.width, ashp.height, Convert.ToBoolean(ashp.frame.flip_h), Convert.ToBoolean(ashp.frame.flip_v))
                }

                print(dir)
            }

        }
        public static double getDirection(float w, float h, bool flipH, bool flipV)
        {
            float endLineX = w * (flipH ? -1 : 1)
            float endLineY = h * (flipV ? -1 : 1)
            float endYAxisX = 0
            float endYAxisY = h
            double angle = (Math.Atan2(endYAxisY, endYAxisX) - Math.Atan2(endLineY, endLineX))
            if (angle < 0) angle += 2 * Math.PI
            return angle * 180.0 / Math.PI
        }
        #ExEnd:ConnectorLineAngle
    }

}

File: examples/src/shapes/connector_line_angle.py
import aspose.slides as slides
import math


def get_direction(w, h, flip_h, flip_v):
    end_line_x = w * (-1 if flip_h else 1)
    end_line_y = h * (-1 if flip_v else 1)
    end_y_axis_x = 0
    end_y_axis_y = h
    angle = math.atan2(end_y_axis_y, end_y_axis_x) - math.atan2(end_line_y, end_line_x)
    if angle < 0:
         angle += 2 * math.pi
    return angle * 180.0 / math.pi


def connector_line_angle(global_opts):
    with slides.Presentation(global_opts.data_dir + "shapes_connector_line_angle.pptx") as pres:
        slide = pres.slides[0]
        for shape in slide.shapes:
            direction = 0.0
            if type(shape) is slides.AutoShape:
                if shape.shape_type == slides.ShapeType.LINE:
                    direction = get_direction(shape.width, shape.Height, shape.frame.flip_h, shape.frame.flip_v)
            elif type(shape) is slides.Connector:
                direction = get_direction(shape.width, shape.height, shape.frame.flip_h, shape.frame.flip_v)
        print(direction)
